tools/codegen/proc_interface_gen/main.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+func withArgs(args []string, errorf func(format string, a ...interface{})) {
+	var outFilePath string
+	var mappings []string
+
+	rootCmd := cobra.Command{
+		Use: "procInterfaceGen <File descriptor set protobuf>",
+		Short: `
+Tool that parses a [Template](http://TODO) and generates go interface for adapters to implement.
+
+Example: procInterfaceGen metricTemplateFileDescriptorSet.pb -o MetricProcessor.go
+`,
+
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) == 0 {
+				errorf("Must specify a file descriptor set protobuf file.")
+			}
+			if len(args) != 1 {
```

Comment:
We can't just wrap a loop around the whole thing and support N templates in one invocation?